subsys/bluetooth/mesh/Kconfig.models

```diff
@@ -277,6 +277,32 @@ endchoice
 endmenu
 endif
 
+menuconfig BT_MESH_LIGHT_CTRL_REG
```

Comment:
Should not be a separate config from `BT_MESH_LIGHT_CTRL_SRV_REG` below as far as I can tell. Putting a `menuconfig` with sub entries right below it, surrounded by `if` does the trick. 

If the intent is to be able to instantiate and use a regulator without the light_ctrl, I think the whole thing would need to be moved out of mesh.

Reply:
That was the intent, yes. Perhaps we should simplify it now, it's easy enough to move it out if we need to as a separate task in the future.

Reply:
Yeah, ideally, I would have liked to have moved it out of mesh before the next release, so we don't have to make a breaking API change in the release when we need it. I usually meet a lot of resistance in the team whenever I suggest doing anything outside subsys/bluetooth/mesh though, so I haven't really pushed it as hard, even though I think it's 100% the right solution. Don't want to scope creep too much on your PR either.

Reply:
I agree with you on this. Where would you suggest putting it? This would solve Mia's doubts about where to put things in the doc hierarchy as well.

Reply:
subsys/regulator makes sense to me, since "controller" is a very overloaded term